### PR TITLE
enh(reader_exception) : shutdown exception updated, replaced msg by m…

### PR DESCRIPTION
**Pull request by jbrouze**
…sg_fmt, replaced msg inheritance by msg_fmt

# Pull Request Template

## Description

Please include a short resume of the changes and what is the purpose of PR. Any relevant information should be added to help:
* **QA Team** (Quality Assurance) with tests.
* **reviewers** to understand what are the stakes of the pull request.

**Fixes** # (issue)

## Type of change

- [ ] Patch fixing an issue (non-breaking change)
- [ ] New functionality (non-breaking change)
- [ ] Breaking change (patch or feature) that might cause side effects breaking part of the Software
- [ ] Updating documentation (missing information, typo...)

## Target serie

- [ ] 18.10.x
- [ ] 19.04.x
- [ ] 19.10.x
- [ ] 20.04.x (master)

<h2> How this pull request can be tested ? </h2>

Please describe the **procedure** to verify that the goal of the PR is matched. Provide clear instructions so that it can be **correctly tested**.

Any **relevant details** of the configuration to perform the test should be added.

#### Community contributors & Centreon team

- [ ] I followed the **coding style guidelines** provided by Centreon
- [ ] I have commented my code, especially new **classes**, **functions** or any **legacy code** modified. (***docblock***)
- [ ] I have commented my code, especially **hard-to-understand areas** of the PR.
- [ ] I have made corresponding changes to the **documentation**.
- [ ] I have **rebased** my development branch on the base branch (master, maintenance).

#### Centreon team only

- [ ] I have made sure that the **unit tests** related to the story are successful.
- [ ] I have made sure that **unit tests cover 80%** of the code written for the story.
- [ ] I have made sure that **acceptance tests** related to the story are successful (**local and CI**)

### CODE REVIEW

File: src/ccb_core/file/splitter.cc [hunk 1]
```diff
@@ -370,11 +359,12 @@ void splitter::_open_write_file() {
...
     std::string file_path(get_file_path(_wid));
-    logging::info(logging::high)
-        << "file: opening new file '" << file_path.c_str() << "'";
+    logging::info(logging::high) << "file: opening new file '"
+                                 << file_path.c_str() << "'";
     try {
       _wfile.reset(new cfile(file_path, fs_file::open_read_write_no_create));
-    } catch (exceptions::msg const& e) {
+    }
+    catch (exceptions::msg const& e) {
```
Comment (bouda1) [inline src/ccb_core/file/splitter.cc:367]: old exception

File: src/ccb_core/processing/feeder.cc [hunk 2]
```diff
@@ -121,8 +120,8 @@ void feeder::start() {
   std::unique_lock<std::mutex> lock(_started_m);
   _stopped = false;
   if (!_client)
-      throw exceptions::msg()
-            << "could not process '" << _name << "' with no client stream";
+    throw exceptions::msg() << "could not process '" << _name
+                            << "' with no client stream";
```
Comment (bouda1) [inline src/ccb_core/processing/feeder.cc:125]: old exception

File: src/ccb_core/file/cfile.cc [hunk 3]
```diff
@@ -175,15 +174,13 @@ long cfile::write(void const* buffer, long size) {
   size_t retval(fwrite(buffer, 1, size, _stream));
   if (ferror(_stream)) {
     char const* msg(strerror(errno));
-    throw exceptions::msg()
-          << "cannot write " << size << " bytes to file: " << msg;
+    throw exceptions::msg() << "cannot write " << size
+                            << " bytes to file: " << msg;
```
[the remark this answers is not in the document]
Comment (jbrouze) [reply src/ccb_core/file/cfile.cc:178]: already changed in ccb core branch

File: src/ccb_core/file/splitter.cc (hunk 1 above)
Comment (jbrouze) [reply src/ccb_core/file/splitter.cc:367]: already changed in ccb core branch

File: src/ccb_core/processing/feeder.cc (hunk 2 above)
Comment (jbrouze) [reply src/ccb_core/processing/feeder.cc:125]: already changed in ccb core branch

File: include/com/centreon/broker/bam/configuration/reader_exception.hh [hunk 4]
```diff
@@ -19,7 +19,8 @@
 #ifndef CCB_CONFIGURATION_READER_EXCEPTION_HH
 #define CCB_CONFIGURATION_READER_EXCEPTION_HH
 
-#include "com/centreon/broker/exceptions/msg.hh"
+#include <exception>
```
Comment (bouda1) [inline include/com/centreon/broker/bam/configuration/reader_exception.hh:22]: Is this really needed?

File: src/50-tcp/connector.cc [hunk 5]
```diff
@@ -23,11 +23,13 @@
 #include <sstream>
 
 #include "com/centreon/broker/exceptions/msg.hh"
```
Comment (bouda1) [inline src/50-tcp/connector.cc:25]: Why keeping this line?

**Pull request merged**